scripts/generateCliCode.ts

```diff
@@ -258,16 +265,23 @@ class Generator {
     }
   }
 
-  protected walkProperty(propertyName: string, preference: yamlObject, settingsTree: settingsTreeType): void {
+  protected walkProperty(
+    propertyName: string,
+    preference: yamlObject,
+    notAvailable: boolean,
+    settingsTree: settingsTreeType): void {
+    const platforms = preference['x-rd-platforms'] ?? [];
+
+    notAvailable ||= platforms.length > 0 && !platforms.includes(os.platform());
```

Comment:
FWIW, `os.platform()` is also available as `process.platform` (a property off a global variable), which means you don't need to import it.